### PR TITLE
BOOM-4362 E2E validation test automation with CI scheduled run

### PR DESCRIPTION
**Pull request by deniz-modelCat**
## Summary
- Adds a comprehensive E2E test suite (82 tests) for `modelcat_validate` CLI using pytest + subprocess, covering all validation scenarios across dataset structure, metadata, COCO annotations, images, cross-split integrity, and more
- Implements a POM-style test framework with `CLIRunner`, `CLIResult`, `DatasetBuilder`, and `DatasetMutators` for clean separation of concerns and maintainability
- Adds a GitHub Actions scheduled workflow (`qa-e2e-scheduled.yml`) running daily at 12:47 UTC with Slack notifications to the QA channel

## Test Coverage (82 tests)

| Group | Tests | What it covers |
|-------|-------|----------------|
| CLI Arguments | 7 | Missing args, valid invocations, flag combinations, verbose mode |
| Dataset Structure | 5 | Non-existent path, missing `dataset_infos.json`, `annotations/`, `images/` |
| dataset_infos.json | 26 | Invalid JSON, missing keys (`splits`, `task_templates`, `description`, `config_name`, `builder_name`), split validation, task type checks, labels, size/count metadata with and without auto-fix |
| COCO Annotations | 24 | Missing/corrupt files, missing sections and fields (`categories`, `images`, `annotations` + subfields), duplicate IDs, orphan references, label mismatches, auto-fix-2 for width/height |
| Image Checks | 5 | Missing images on disk, images without annotations, duplicate filenames, empty categories |
| Cross-Split | 3 | Train/test leakage, test/validation leakage, single-image leakage detection |
| Thumbnail | 3 | Auto-generation, no-images edge case, pre-existing thumbnail |
| Signature & Output | 4 | Signature hash, log file creation, summary sections |
| Backup | 3 | Auto-fix backup directory, backup messages, COCO file backup |
| Happy Paths | 5 | Classification, object detection, keypoints, all flags combined, verbose |

## Known Product Bugs Exposed
10 tests intentionally fail, exposing 2 product bugs (tracked separately):
1. **Unprotected `pycocotools.COCO()` calls** – helper functions crash before `param_check` validates the data (9 tests)
2. **Unprotected `json.load()` in `validate_split_sizes`** – crashes on corrupted `dataset_infos.json` (1 test)

## Reporting

# Slack notification example:
https://modelcat.slack.com/archives/C0AMK1P30HE/p1778493567671029
image

# The pipeline generates an HTML report: 
image

## How to Run
```bash
# All E2E tests
python -m pytest tests/e2e/ -m e2e -v

# Specific test file
python -m pytest tests/e2e/tests/test_coco_validation.py -v

# Only passing tests (exclude known bugs)
python -m pytest tests/e2e/ -m e2e -v -k "not (invalid_json or missing_on_disk or missing_images_section or missing_category_id or missing_image_id or missing_image_filename or missing_annotation_id or missing_annotation_image_id or missing_annotation_category_id)"

### CODE REVIEW

File: .github/workflows/qa-e2e-scheduled.yml [hunk 1]
```diff
@@ -0,0 +1,265 @@
...
+#   Create at api.slack.com/apps → Your App → OAuth & Permissions → Bot User OAuth Token.
+#   Scopes: chat:write, chat:write.public. Invite the bot to channel C0AMK1P30HE (or it must have chat:write.public).
+name: QA ModelCat Connector E2E scheduled
+
+on:
+  push:
+  pull_request:
+  workflow_dispatch:
+  schedule:
+    - cron: '47 12 * * *'   # 12:47 UTC daily (30 min after Playwright run)
```
Comment (vedin-eta) [inline .github/workflows/qa-e2e-scheduled.yml]: @jeremi-eta I assume modelcat E2E tests daily might be an overkill, we rarely update this repo. Maybe weekly? or every 3 days?

**Review by vedin-eta — COMMENTED** [review]
This looks good to me, need to check the frequency of the tests first. Also, before merging it all, I'd like to see one run on your fork. If some tests are failing, we can fix them first.

File: .github/workflows/qa-e2e-scheduled.yml (hunk 1 above)
Comment (deniz-modelCat) [reply .github/workflows/qa-e2e-scheduled.yml]: Will be on merge and every Monday. Updated

**Review by vedin-eta — APPROVED** [review]
